### PR TITLE
[release/6.0] Backport consolidation of Newtonsoft versions

### PR DESCRIPTION
**Pull request by steveisok**
Backport of:
https://github.com/dotnet/runtime/pull/71402
https://github.com/dotnet/runtime/pull/72662
https://github.com/dotnet/runtime/pull/72451

```
Newtonsoft.Json => v13.0.1
Newtonsoft.Json.Bson => v1.0.2
```

### CODE REVIEW

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonTestHelper.cs
Comment (teo-tsirpanis) [inline src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonTestHelper.cs:372]:
```suggestion
                newtonsoft = new JsonTextReader(new StringReader(jsonString)) { MaxDepth = null };
```

**Comment by jakobbotsch** [comment]
There were some follow ups to this one in #72451 and #72662.

**Comment by steveisok** [comment]
@teo-tsirpanis Can you give this another review?

@carlossanlop this is infra / testing changes and I don't think needs to route through tactics.  Can this go in once approved?

**Review by teo-tsirpanis — APPROVED** [review]
CI is green, so…

**Review by carlossanlop — APPROVED** [review]
LGTM assuming CI is green.

**Pull request merged**
